src/shims/unix/solarish/foreign_items.rs

```diff
@@ -69,6 +69,31 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                 this.write_null(dest)?;
             }
 
+            "pset_info" if this.frame_in_std() => {
+                let [pset, tpe, cpus, list] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
```

Comment:
Please write this the way we write all our shims: with a line like `let pset = this.read_scalar(pset)?.to_i32()?;` etc for each argument. The "turning arguments into a typed representation" should be separate from all further processing.